### PR TITLE
Start Documenting  BP REST API v2

### PR DESCRIPTION
**Pull request by imath**
Adds the following page to developer doc:
- Introduction about the BP REST API
- Reference: list of REST API Routes
- Components routes
- Members routes

Working on these docs made me suggesting improvements about Components and Members REST controllers, @renatonascalves could you have a look at these?

See https://github.com/buddypress/bp-documentation/issues/298

Ticket: https://buddypress.trac.wordpress.org/ticket/9145

---
**This Pull Request is for code review only. Please keep all other discussion in the BuddyPress Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the WordPress Core Handbook for more details.**

### CODE REVIEW

File: docs/developer/execution-contexts/rest-api/README.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# BuddyPress REST API (v2)
+
+> [!IMPORTANT]
+> This is the documentation for the BP REST API **v2**. This version was introduced in BuddyPress 15.0.0. The BP REST API v1 is deprecated as of BuddyPress 15.0.0. If you still need to use v1, you can download and activate the archived [BP REST plugin](https://github.com/buddypress/BP-REST) and refer to this [documentation](https://developer.buddypress.org/bp-rest-api/). We strongly advise you to update your code to support **v2** as soon as you can.
```
Comment (renatonascalves) [inline docs/developer/execution-contexts/rest-api/README.md:4]:
```suggestion
> This is the documentation for the BP REST API **v2**. This version was introduced in BuddyPress 15.0.0. The BP REST API v1 is deprecated as of BuddyPress 15.0.0. If you still need to use v1, you can download and activate the archived [BuddyPress RESTful API plugin](https://github.com/buddypress/BP-REST) and refer to this [documentation](https://developer.buddypress.org/bp-rest-api/). We strongly advise you to update your code to support **v2** as soon as you can.
```

Comment (renatonascalves) [reply docs/developer/execution-contexts/rest-api/README.md:4]: Based on your recent comment [here](https://wordpress.slack.com/archives/C02RQBYUG/p1727896648522939).

**Review by renatonascalves — CHANGES_REQUESTED** [review]

File: src/bp-core/classes/class-bp-core-components-rest-controller.php [hunk 2]
```diff
@@ -552,10 +552,11 @@ public function get_item_schema() {
 						'type'        => 'string',
 					),
 					'features'    => array(
-						'description' => __( 'Information about active features for the component.', 'buddypress' ),
-						'type'        => 'array',
```
[the remark this answers is not in the document]
Comment (renatonascalves) [reply src/bp-core/classes/class-bp-core-components-rest-controller.php:556]: The change looks good, but it looks to me the type of the data being returned is not an `object`.

**Comment by renatonascalves** [comment]
There is one unit test failing due to a flaky unit test. I've been trying to identify why it fails sometimes only. You can usually rerun the failed action to "fix" it. While I try to come up with a _final_ fix.

CleanShot 2024-10-07 at 10 03 23@2x

File: src/bp-core/classes/class-bp-core-components-rest-controller.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (imath) [reply src/bp-core/classes/class-bp-core-components-rest-controller.php:556]: I agree in PHP a non numeric associative array is sent by the controller. But as it's a non numeric associative array, the JSON response treat it as an object:
 
Capture d’écran 2024-10-08 à 04 55 16

I'm wondering, should we consider the JSON response or the PHP part to type schema properties?

[the remark this answers is not in the document]
Comment (renatonascalves) [reply src/bp-core/classes/class-bp-core-components-rest-controller.php:556]: Interesting.

[the remark this answers is not in the document]
Comment (renatonascalves) [reply src/bp-core/classes/class-bp-core-components-rest-controller.php:556]: I think we should consider the JSON response, that's what the REST API consumers will see. Not the PHP version of the data.

**Review by renatonascalves — APPROVED** [review]

**Pull request closed without merging**